### ISSUE TITLE
[BUG/Help] <title> P-tuning脚本运行loss 一直为0 ？

### ISSUE THREAD

**Issue by valkryhx** [THUDM/ChatGLM2-6B#200]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

P-tuning一个1000条数据集，loss一直为0，何解？没有用load_in_int8 和quan4 。

### Steps To Reproduce

如题。

### Environment

```markdown
- OS:linux
- Python:3.10
- Transformers:4.27.1
- PyTorch:2.0.0
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by xiuxiulu** [comment]
怀疑模型结构上就没加上前缀，暂时还是用LoRA吧

**Comment by valkryhx** [comment]
> 怀疑模型结构上就没加上前缀，暂时还是用LoRA吧


请问 有没有LORA 微调chatglm2的github项目推荐，我之前看的medicalGPT项目昨天还SFT没问题，今天看到chatglm2-6b的modeling_chatglm.py更新之后 ，多卡微调就开始报错了。 您方便推荐下LORA微调chatglm2的项目的话，那就感激不尽。

**Comment by valkryhx** [comment]
好的 我试试

**Comment by lilongxian** [comment]
learning rate 调小一点，另外如果GPU够的话可以把累加批适当增大，这两个（数值相乘==真实batch_szie）：--per_device_train_batch_size 1   --gradient_accumulation_steps 16

**Comment by asjmasjm** [comment]
I faced the same problem with this script using A10@40G * 5:
PRE_SEQ_LEN=64
LR=1e-3
CUDA_VISIBLE_DEVICES=0,1,2,3,4 python -m torch.distributed.launch --nproc_per_node=5 main.py \
    --do_train \
    --train_file ../AdvertiseGen/train.json \
    --validation_file ../AdvertiseGen/dev.json \
    --preprocessing_num_workers 10 \
    --prompt_column content \
    --response_column summary \
    --overwrite_cache \
    --model_name_or_path chatglm2-6b \
    --output_dir adgen-chatglm2-6b-pt-$LR \
    --overwrite_output_dir \
    --max_source_length 64 \
    --max_target_length 64 \
    --per_device_train_batch_size 128 \
    --per_device_eval_batch_size 8  \
    --gradient_accumulation_steps 16 \
    --predict_with_generate \
    --max_steps 200 \
    --logging_steps 1 \
    --save_steps 50 \
    --learning_rate $LR \
    --pre_seq_len $PRE_SEQ_LEN

the loss has been seemingly back to the normal track after PRE_SEQ_LEN, max_source_length and max_target_length are set to 128 or 256 while per_device_train_batch_size set to 64 or 32 accordingly to avoid OOM. 
Still no idea why the loss-equal-0 problem has been related to these hyper parameters...
